builder/vsphere/supervisor/builder.go

```diff
@@ -57,6 +57,22 @@ func (b *Builder) Run(ctx context.Context, ui packersdk.Ui, hook packersdk.Hook)
 		&StepValidatePublish{
 			Config: &b.config.ValidatePublishConfig,
 		},
+	)
+
+	// conditionally add steps to validate import spec and import images from source URL as VM image.
+	if b.config.ValidateImportConfig.ImportSourceURL != "" {
+		steps = append(steps,
+			&StepValidateImport{
+				Config: &b.config.ValidateImportConfig,
+			},
```

Comment:
After looking at the implementation of this step, I kind of feel like we may not need it in this plugin. Most of its functionality duplicates the CRD scheme check and webhook validation. I am concerned that if we change those in the future (which is likely when we add ISO import support, for example), we would need to update this step as well.

The reason we have the `ValidatePublish` step is that a VM is published at the end, which could lead to a poor user experience if errors in their publish configs are discovered after the VM is deployed and customized. For import, since it's executed at a pretty early stage, I think it's sufficient to perform some basic config checks in the `Prepare()` function and let the image-registry-operator controller and webhook handle the rest. The Packer plugin could simply output a message if any error occurs.

Reply:
I think this is a fair point, we don't have to maintain 2 steps, I'm combining them into one step instead, and make it easier to maintain.